src/GafferScene/TransformQuery.cpp

```diff
@@ -324,40 +332,48 @@ void TransformQuery::compute( Gaffer::ValuePlug* const output, Gaffer::Context c
 			ScenePlug const* const splug = scenePlug();
 			assert( splug != 0 );
 
-			ScenePlug::ScenePath const path = ScenePlug::stringToPath( loc );
+			Gaffer::BoolPlug const* const eplug = splug->existsPlug();
+			assert( eplug != 0 );
+
+			// NOTE : scene exists plug returns true by default when there is no input scene
 
-			if( splug->exists( path ) )
+			if( eplug->source< Gaffer::Plug >() != eplug )
 			{
-				switch( static_cast< Space >( spacePlug()->getValue() ) )
+				ScenePlug::ScenePath const path = ScenePlug::stringToPath( loc );
+
+				if( splug->exists( path ) )
 				{
-					case Space::Local:
-						m = splug->transform( path );
-						break;
-					case Space::World:
-						m = splug->fullTransform( path );
-						break;
-					case Space::Relative:
+					switch( static_cast< Space >( spacePlug()->getValue() ) )
 					{
-						std::string const rloc = relativeLocationPlug()->getValue();
-
-						if( ! rloc.empty() && ( rloc != loc ) ) // NOTE : If loc == rloc then m is identity
+						case Space::Local:
+							m = splug->transform( path );
+							break;
+						case Space::World:
+							m = splug->fullTransform( path );
+							break;
+						case Space::Relative:
 						{
-							ScenePlug::ScenePath const rpath = ScenePlug::stringToPath( rloc );
+							std::string const rloc = relativeLocationPlug()->getValue();
 
-							if( splug->exists( rpath ) )
+							if( ! rloc.empty() && ( rloc != loc ) ) // NOTE : If loc == rloc then m is identity
```

Comment:
This could produce some unexpected behaviour if rloc and loc represent the same path, but with a different string representation - for example "/sphere" and "/sphere/" are both valid strings referring to the same path.  I guess this would just result in getting a matrix that is close to identity, down to floating point precision, but not necessarily exactly identity - not really sure if this an issue, it would be kinda sloppy of the user to express the same path differently - but maybe it would be worth comparing `rpath` and `path` instead, for a more reliable comparison.

Reply:
fixed with additional test cases

Reply:
Looks good